### PR TITLE
use condition attributes to specify Python 2 and 3 dependencies

### PR DESCRIPTION
**Pull request by dirk-thomas**
Similar to ros/catkin#1025.

### CODE REVIEW

File: test/test_rosbag/package.xml [hunk 1]
```diff
@@ -7,36 +11,39 @@
...
   <buildtool_depend version_gte="0.5.68">catkin</buildtool_depend>
 
   <build_depend>boost</build_depend>
   <build_depend>bzip2</build_depend>
   <build_depend>cpp_common</build_depend>
   <build_depend>message_generation</build_depend>
-  <build_depend>python-imaging</build_depend>
-  <build_depend>rosbag</build_depend>
+  <build_depend condition="$ROS_PYTHON_VERSION == 2">python-imaging</build_depend>
+  <build_depend condition="$ROS_PYTHON_VERSION == 3">python3-imaging</build_depend>
```
Comment (sloretz) [inline test/test_rosbag/package.xml]: I think the equivalent python3 key is [`python3-pil`](https://github.com/ros/rosdistro/blob/0ea30fed0bcb564b9e4d5b2d2efd8783e8e3afa9/rosdep/python.yaml#L5107). There's no `python3-imaging` key (nor is there a debian package with that name).

File: tools/rosbag/package.xml [hunk 2]
```diff
@@ -9,38 +13,39 @@
...
   <buildtool_depend version_gte="0.5.78">catkin</buildtool_depend>
 
-  <build_depend>boost</build_depend>
   <build_depend>cpp_common</build_depend>
-  <build_depend>python-imaging</build_depend>
-  <build_depend>rosbag_storage</build_depend>
-  <build_depend>rosconsole</build_depend>
-  <build_depend>roscpp</build_depend>
+  <build_depend condition="$ROS_PYTHON_VERSION == 2">python-imaging</build_depend>
+  <build_depend condition="$ROS_PYTHON_VERSION == 3">python3-imaging</build_depend>
```
Comment (sloretz) [inline tools/rosbag/package.xml]: `python3-pil` here too

File: tools/rosbag/package.xml [hunk 3]
```diff
@@ -9,38 +13,39 @@
...
-  <run_depend>roscpp</run_depend>
-  <run_depend>roslib</run_depend>
-  <run_depend>rospy</run_depend>
-  <run_depend>std_srvs</run_depend>
-  <run_depend>topic_tools</run_depend>
-  <run_depend>xmlrpcpp</run_depend>
+  <exec_depend>genmsg</exec_depend>
+  <exec_depend>genpy</exec_depend>
+  <exec_depend condition="$ROS_PYTHON_VERSION == 2">python-crypto</exec_depend>
+  <exec_depend condition="$ROS_PYTHON_VERSION == 3">python3-crypto</exec_depend>
```
Comment (sloretz) [inline tools/rosbag/package.xml]: There's no `python3-crypto` key, and the project is dead. [`python3-cryptography`](https://github.com/ros/rosdistro/blob/0ea30fed0bcb564b9e4d5b2d2efd8783e8e3afa9/rosdep/python.yaml#L4925-L4929) is a possible replacement #1599 https://github.com/ros/ros_comm/pull/1606

**Review by sloretz — CHANGES_REQUESTED** [review]

File: test/test_rosbag/package.xml (hunk 1 above)
Comment (dirk-thomas) [reply test/test_rosbag/package.xml]: Good catch: 85394ffe89f981469bbd0954d1483382aff20df8.

File: tools/rosbag/package.xml (hunk 2 above)
Comment (dirk-thomas) [reply tools/rosbag/package.xml]: Good catch: 85394ffe89f981469bbd0954d1483382aff20df8.

File: tools/rosbag/package.xml (hunk 3 above)
Comment (dirk-thomas) [reply tools/rosbag/package.xml]: Alternative PRs: #1609 or #1793. Currently I am leaning towards the former one due to the much simpler diff.

Comment (dirk-thomas) [reply tools/rosbag/package.xml]: With #1609 merged I rebased this patch and updated the dependency to `python(3)-pycryptodome`.

**Review by sloretz — APPROVED** [review]

**Pull request merged**